app/models/source.rb

```diff
@@ -44,63 +44,31 @@ class Source < ActiveRecord::Base
   # without setting the reimport flag. If someone really wants to turn off
   # reimporting, they should edit the source.
   def self.find_or_create_from(attrs={})
-    if attrs && attrs[:url]
-      source = Source.find_or_create_by_url(attrs[:url])
-      attrs.each_pair do |key, value|
-        if key.to_sym == :reimport
-          source.reimport = true if ! source.reimport && value
-        else
-          source.send("#{key}=", value) if source.send(key) != value
-        end
-      end
-      source.save if source.changed?
-      return source
-    else
-      return Source.new(attrs)
-    end
+    return new(attrs) unless attrs[:url]
+
+    source = find_or_create_by_url(attrs[:url])
+    source.reimport = true if attrs.delete(:reimport)
+    source.attributes = attrs
+    source.save if source.changed?
+    source
   end
 
   # Create events for this source. Returns the events created. URL must be set
   # for this source for this to work.
   def create_events!(opts={})
-    cutoff = Time.now.yesterday # All events before this date will be skipped
-    events = []
-    self.to_events(opts).each do |event|
-      if opts[:skip_old]
-        next if event.title.blank? && event.description.blank? && event.url.blank?
-        next if event.old?
-      end
-
-      # Skip invalid events that start after they end
-      next if event.end_time && event.end_time < event.start_time
-
-      # convert to local time, because time zone is simply discarded when event is saved
-      event.start_time.localtime
-      event.end_time.localtime if event.end_time
-
-      # clear duplicate_of_id field in case to_events picked up orphaned duplicate
```

Comment:
Okay, I looked into this some more, and I'm pretty sure we can safely leave this whole clause deleted, because 1. turns out there is already a test-case in place that addresses the bug you were referring to, and 2. the clause is effectively a no-op.

1. Found an existing test that ensures the correct behavior, and it is green with and without this clause: https://github.com/calagator/calagator/pull/272/files#diff-bf219e006cd78e831f27f4c38b306c59L97

2. So, this clause nils out the event and venue `duplicate_of_id` column, if its present. However, `event_or_duplicate` and `venue_or_duplicate` both ensure that we have the progenitor in the case of any dupe, and the progenitor is defined by having a nil `duplicate_of_id` column. Therefore, the event and venue `duplicate_of_id` columns will always be nil, and this clause is a no-op.

Good to merge?